monai/data/utils.py

```diff
@@ -517,13 +519,14 @@ def zoom_affine(affine: np.ndarray, scale: Sequence[float], diagonal: bool = Tru
     if len(affine) != len(affine[0]):
         raise ValueError(f"affine must be n x n, got {len(affine)} x {len(affine[0])}.")
     scale_np = np.array(scale, dtype=float, copy=True)
-    if np.any(scale_np <= 0):
-        raise ValueError("scale must contain only positive numbers.")
+
     d = len(affine) - 1
+    # compute original pixdim
+    norm = np.sqrt(np.sum(np.square(affine), 0))[:-1]
     if len(scale_np) < d:  # defaults based on affine
-        norm = np.sqrt(np.sum(np.square(affine), 0))[:-1]
         scale_np = np.append(scale_np, norm[len(scale_np) :])
-    scale_np = scale_np[:d]
```

Comment:
If we remove this line, when `len(scale_np) > d`, error happens in `fall_back_tuple(scale_np, norm)`

Reply:
Thanks for pointing this, updated in the latest commit.

Reply:
And BTW, as we require `length of pixdim = input size`, there should be no case that `len(scale_np) > d` anymore.

Thanks.